### PR TITLE
Add Routes

### PR DESCRIPTION
**Pull request by RawAlsaadi**

### CODE REVIEW

**Comment by RawAlsaadi** [comment]
I'll add authentication and role based access in another PR

File: api/main.go [hunk 1]
```diff
@@ -30,16 +31,26 @@ func main() {
...
-
-	r.GET("/roles", GetRolesHandler(db))
+	r.GET("/roles", Handlers.GetRolesHandler(db))
+	r.GET("/roles/:id", Handlers.GetRoleByIdHandler(db))
 
-	r.POST("/user", CreateUserHandler(db))
+	r.POST("/user", Handlers.CreateUserHandler(db))
+	r.GET("/users", Handlers.GetUsersHandler(db))
+	r.GET("/users/:id", Handlers.GetUserByIdHandler(db))
+	r.GET("/users/email/:email", Handlers.GetUserByEmailHandler(db))
```
Comment (noleu) [inline api/main.go]: route should be /users/:email

File: api/main.go [hunk 2]
```diff
@@ -30,16 +31,26 @@ func main() {
...
+	r.PUT("/users/:id", Handlers.UpdateUserHandler(db))
 
-	r.GET("/users", GetUserHandler(db))
+	r.POST("/events", Handlers.CreateEventHandler(db))
+	r.GET("/events", Handlers.GetEventsHandler(db))
+	r.GET("/events/:id", Handlers.GetEventByIdHandler(db))
+	r.GET("/events/user/:userId", Handlers.GetEventsByUserIdHandler(db))
```
Comment (noleu) [inline api/main.go]: route should be /events/:userId

File: api/Models/UpdateEventInput.go
Comment (noleu) [inline api/Models/UpdateEventInput.go:1]: Can we move this to a separate folder, this folder should only contain the db model

File: api/Models/UpdateUserInput.go
Comment (noleu) [inline api/Models/UpdateUserInput.go:1]: Can we move this to a separate folder, this folder should only contain the db model

File: api/main.go [hunk 3]
```diff
@@ -30,16 +31,26 @@ func main() {
...
 	InitializeRoles(db)
 
-	r := gin.Default()
-
-	r.GET("/roles", GetRolesHandler(db))
+	r.GET("/roles", Handlers.GetRolesHandler(db))
+	r.GET("/roles/:id", Handlers.GetRoleByIdHandler(db))
 
-	r.POST("/user", CreateUserHandler(db))
+	r.POST("/user", Handlers.CreateUserHandler(db))
```
Comment (qimolin) [inline api/main.go:42]: We can move create user to a separate /auth/register route when implementing the auth.

File: api/Models/UpdateEventInput.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package Models
+
+import "time"
+
+type UpdateEventInput struct {
```
Comment (qimolin) [inline api/Models/UpdateEventInput.go]: What is this model for?

File: api/Models/UpdateUserInput.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package Models
+
+type UpdateUserInput struct {
```
Comment (qimolin) [inline api/Models/UpdateUserInput.go]: What is this model for?

File: api/Models/UpdateEventInput.go (hunk 4 above)
Comment (RawAlsaadi) [reply api/Models/UpdateEventInput.go]: It is a separate struct to represent the fields that can be updated using the PUT endpoint (Everything can be updated except the id)

**Review by qimolin — APPROVED** [review]

**Pull request merged**